build.sbt

```diff
@@ -462,6 +476,210 @@ lazy val spark = (project in file("spark"))
     Antlr4 / antlr4GenListener := true,
     Antlr4 / antlr4GenVisitor := true,
 
+    // Hack to avoid errors related to missing repo-root/target/scala-2.12/classes/
+    createTargetClassesDir := {
+      val dir = baseDirectory.value.getParentFile / "target" / "scala-2.12" / "classes"
+      Files.createDirectories(dir.toPath)
+    },
+    Compile / compile := ((Compile / compile) dependsOn createTargetClassesDir).value,
+    // Generate the package object to provide the version information in runtime.
+    Compile / sourceGenerators += Def.task {
+      val file = (Compile / sourceManaged).value / "io" / "delta" / "package.scala"
+      IO.write(file,
+        s"""package io
+           |
+           |package object delta {
+           |  val VERSION = "${version.value}"
+           |}
+           |""".stripMargin)
+      Seq(file)
+    },
+  )
+
+// ============================================================
+// Module 2: sparkV1Shaded (v1 without DeltaLog for v2 dependency)
+// ============================================================
+lazy val sparkV1Shaded = (project in file("spark-v1-shaded"))
```

Comment:
sparkV1Filtered

Reply:
and please add a short comment explaining why we are doing this.